pkg/s3wrapper/client.go

```diff
@@ -74,28 +79,53 @@ type Config struct {
 	S3Bucket           string `envconfig:"S3_BUCKET"`
 	AwsAccessKeyID     string `envconfig:"AWS_ACCESS_KEY_ID"`
 	AwsSecretAccessKey string `envconfig:"AWS_SECRET_ACCESS_KEY"`
+
+	// Warning - the files stored in this bucket are publicly viewable and therefore
+	// should only be used for storing RHCOS image files that are readily available on the Internet
+	PublicS3EndpointURL      string `envconfig:"S3_ENDPOINT_URL_PUBLIC"`
+	PublicRegion             string `envconfig:"S3_REGION_PUBLIC"`
+	PublicS3Bucket           string `envconfig:"S3_BUCKET_PUBLIC"`
+	PublicAwsAccessKeyID     string `envconfig:"AWS_ACCESS_KEY_ID_PUBLIC"`
+	PublicAwsSecretAccessKey string `envconfig:"AWS_SECRET_ACCESS_KEY_PUBLIC"`
 }
 
 const timestampTagKey = "create_sec_since_epoch"
 
+var BootFileExtensions = [...]string{"iso", "initrd.img", "rootfs.img", "vmlinuz"}
+var ISOFileTypes = map[string]string{
+	"initrd.img": "/images/pxeboot/initrd.img",
+	"rootfs.img": "/images/pxeboot/rootfs.img",
+	"vmlinuz":    "/images/pxeboot/vmlinuz",
+}
+
 // NewS3Client creates new s3 client using default config along with defined env variables
 func NewS3Client(cfg *Config, logger logrus.FieldLogger) *S3Client {
-	awsSession, err := newS3Session(cfg)
+	awsSession, err := newS3Session(cfg.AwsAccessKeyID, cfg.AwsSecretAccessKey, cfg.Region, cfg.S3EndpointURL)
 	if err != nil {
 		logger.WithError(err).Error("failed to create s3 session")
 		return nil
 	}
-
 	client := s3.New(awsSession)
 	if client == nil {
 		return nil
 	}
 
-	isoUploader := NewISOUploader(logger, client, cfg.S3Bucket)
-	return &S3Client{client: client, session: awsSession, cfg: cfg, log: logger, isoUploader: isoUploader}
+	publicAwsSession, err := newS3Session(cfg.PublicAwsAccessKeyID, cfg.PublicAwsSecretAccessKey, cfg.PublicRegion, cfg.PublicS3EndpointURL)
+	if err != nil {
+		return nil
+	}
+	publicClient := s3.New(awsSession)
+	if client == nil {
```

Comment:
copy error? 
```suggestion
	if publicClient == nil {
```
And suggesting to add a log